src/js/actions/staff.js

```diff
@@ -21,6 +24,37 @@ const keyOffDate = (availableSlots) => {
     return formattedSlots;
 };
 
+export const fetchSlots = startDate =>
+    (dispatch, getState) => {
+        const month = startDate.month();
+        const anyResourceId = Object.keys(getState().staff.allAvailableSlots)[0];
```

Comment:
I only did the first one but it could by any resource id. It's one way of checking if I have fetched data for that month for all the resources. Alternatively, I could keep a state of months I have fetched. I just realized that this method won't work for spanning multiple years...